src/coreclr/vm/jithelpers.cpp

```diff
@@ -634,72 +687,59 @@ HCIMPL1_V(UINT64, JIT_Dbl2ULngOvf, double val)
 }
 HCIMPLEND
 
-
-#if !defined(TARGET_X86) || defined(TARGET_UNIX)
-
-HCIMPL1_V(INT64, JIT_Dbl2Lng, double val)
+HCIMPL1_V(UINT32, JIT_Dbl2UInt, double val)
 {
     FCALL_CONTRACT;
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
+    const double uint_max = 4294967295.0;
+    return (val != val || val <= 0) ? 0 : (val >= uint_max) ? UINT32_MAX : (UINT32)val;
 
-    return((INT64)val);
+#else
+    return((UINT32)val);
+
+#endif //TARGET_X86 || TARGET_AMD64
 }
 HCIMPLEND
 
-HCIMPL1_V(int, JIT_Dbl2IntOvf, double val)
+/*********************************************************************/
+HCIMPL1_V(INT32, JIT_Dbl2Int, double val)
 {
     FCALL_CONTRACT;
-
-    const double two31 = 2147483648.0;
-
-        // Note that this expression also works properly for val = NaN case
-    if (val > -two31 - 1 && val < two31)
-        return((INT32)val);
-
-    FCThrow(kOverflowException);
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
+    const double int32_min = -2147483648.0;
+    const double int32_max_plus_1 = 2147483648.0;
+    return (val!= val) ? 0 : (val <= int32_min) ? INT32_MIN : (val >= int32_max_plus_1) ? INT32_MAX : (INT32)val;
+#else
+    return((INT32)val);
+#endif // TARGET_X86 || TARGET_AMD64
 }
 HCIMPLEND
 
-HCIMPL1_V(INT64, JIT_Dbl2LngOvf, double val)
+/*********************************************************************/
+HCIMPL1_V(UINT64, JIT_Dbl2ULng, double val)
 {
     FCALL_CONTRACT;
 
-    const double two63  = 2147483648.0 * 4294967296.0;
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
+    const double uint64_max_plus_1 = 4294967296.0 * 4294967296.0;
+    return (val != val || val < 0) ? 0 : (val >= uint64_max_plus_1) ? UINT64_MAX : (UINT64)val;
 
-    // Note that this expression also works properly for val = NaN case
-    // We need to compare with the very next double to two63. 0x402 is epsilon to get us there.
-    if (val > -two63 - 0x402 && val < two63)
-        return((INT64)val);
-
-    FCThrow(kOverflowException);
-}
-HCIMPLEND
-
-#ifndef TARGET_WINDOWS
-namespace
-{
-    bool isnan(float val)
-    {
-        UINT32 bits = *reinterpret_cast<UINT32*>(&val);
-        return (bits & 0x7FFFFFFFU) > 0x7F800000U;
-    }
-    bool isnan(double val)
-    {
-        UINT64 bits = *reinterpret_cast<UINT64*>(&val);
-        return (bits & 0x7FFFFFFFFFFFFFFFULL) > 0x7FF0000000000000ULL;
-    }
-    bool isfinite(float val)
-    {
-        UINT32 bits = *reinterpret_cast<UINT32*>(&val);
-        return (~bits & 0x7F800000U) != 0;
+#else
```

Comment:
The `#else` path cannot possibly implement the saturating behavior that we are standardizing on. It should be deleted.

Reply:
https://github.com/khushal1996/runtime/blob/f3e4bf5556f730dec35c8951aa97ba9480f5413b/src/coreclr/jit/morph.cpp#L626

The ```#else``` was kept as is because ```arm32``` still depends on it and is not saturating. In this PR, we are standardizing saturating behavior for ```x86/x64``` and not all hardware. This PR was specific to Intel Architecture and hence behavior on all other architectures were kept as is. The details being similar to my comment here --> https://github.com/dotnet/runtime/pull/97529#discussion_r1534702884

Reply:
It makes the unnecessarily more complicated and harder to review. I do not think keeping arm32 absolutely intact should be a goal for this PR.

Reply:
Sure. Thanks for pointing that out. I have started a conversation here https://github.com/dotnet/runtime/pull/97529#discussion_r1534728653 to get a consensus. We shall resolve it soon.